ggml/src/ggml-vulkan/ggml-vulkan.cpp

```diff
@@ -5951,6 +6098,14 @@ static void ggml_vk_mul_mat_vec_q_f16(ggml_backend_vk_context * ctx, vk_context&
             ctx->prealloc_y_last_tensor_used = src1;
         }
     }
+    if (quantize_y) {
+        if (ctx->prealloc_y_last_pipeline_used != to_q8_1.get() ||
+            ctx->prealloc_y_last_tensor_used != src1) {
+            ggml_vk_quantize_q8_1(ctx, subctx, { d_Qy, qy_buf_offset, VK_WHOLE_SIZE }, { d_Y, 0, VK_WHOLE_SIZE }, y_ne * ne12 * ne13, true);
```

Comment:
The "prealloc_y_need_sync" logic is needed for quantize_y, too. I think it would make sense to pull in the change from https://github.com/ggml-org/llama.cpp/pull/15544, it may help performance here.